integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchemaCommon.scala

```diff
@@ -275,6 +275,56 @@ class AlterTableColumnSchemaGenerator(
         allColumns ++= Seq(columnSchema)
       }
       newCols ++= Seq(columnSchema)
+      if(field.dataType== Some("Array") ) {
+        columnSchema.setNumberOfChild(field.children.size)
+        val childField = field.children.get(0)
+        val childSchema: ColumnSchema = TableNewProcessor.createColumnSchema(
+          childField,
```

Comment:
@akkio-97 i think here once you added column, the column will be at last, i mean to say like below

existing dim + new added dim + existing measures + new measures, same has to be present in thrift column schema also, you can refer the existing test cases to know better.

Also, have a test cases, with sort column, measures in sort column, dim in sort column etc,

please recheck and correct this and confirm.

Reply:
The order is maintained like above. The thrift column schema does not contain new columns. 
The new dimensions(including complex) are added after the old dimensions only. After which all measure columns are added.

Reply:
then how come when i debug your test case, i get the wrong order? 
![image](https://user-images.githubusercontent.com/21074821/114054749-c235c500-98ad-11eb-9ddb-9700446d449d.png)

Reply:
@akkio-97 , my bad, i debugged a test case where there were no dimensions

Reply:
Right, thrift is updated.

Reply:
Sorting is not supported for complex columns